Cache/IO.lean

```diff
@@ -154,34 +156,28 @@ where
     else
       loop h (← processLine a line)
 
-/-- Runs a terminal command and retrieves its output -/
-def runCmd (cmd : String) (args : Array String) (throwFailure := true) : IO String := do
-  let out ← IO.Process.output { cmd := cmd, args := args }
-  if out.exitCode != 0 && throwFailure then throw <| IO.userError out.stderr
-  else return out.stdout
-
 def runCurl (args : Array String) (throwFailure := true) : IO String := do
-  runCmd (← getCurl) args throwFailure
+  runCmdWithInput (← getCurl) args (throwFailure := throwFailure)
```

Comment:
Why is it better to pipe in the empty string rather than use `stdin := .null` as we were before?

Reply:
I thought it was simply removing some code duplication of a function that has been moved from mathlib to std recently, but in the meantime it has been even changed a little further in Mathlib, so probably I'll just close this.